src/cloud/config_manager/controllers/server.go

```diff
@@ -217,12 +219,29 @@ func (s *Server) GetConfigForVizier(ctx context.Context,
 	}
 	AddDefaultTableStoreSize(tmplValues.PEMMemoryRequest, tmplValues.CustomPEMFlags)
 
-	// Next we inject any feature flags that we want to set for this org.
-	orgID, err := s.getOrgIDForDeployKey(tmplValues.DeployKey)
+	// Attempt to get the org ID from DeployKey, otherwise from the Vizier.
+	getOrgID := true
+	var orgID uuid.UUID
+	orgID, err = s.getOrgIDForDeployKey(tmplValues.DeployKey)
 	if err != nil || orgID == uuid.Nil {
-		log.WithError(err).Error("Error getting org ID from deploy key, skipping feature flag logic")
-	} else {
+		log.WithError(err).Error("Error getting org ID from deploy key")
+		getOrgID = false
+	}
+	if !getOrgID && in.VizierID != "" {
```

Comment:
Instead of tracking a separate `getOrgID` var you can just check if `orgID == uuid.Nil`.